packages/datadog-instrumentations/src/router.js

```diff
@@ -169,11 +169,103 @@ function createWrapRouterMethod (name) {
 
 const wrapRouterMethod = createWrapRouterMethod('router')
 
-addHook({ name: 'router', versions: ['>=1'] }, Router => {
+addHook({ name: 'router', versions: ['>=1 <2'] }, Router => {
   shimmer.wrap(Router.prototype, 'use', wrapRouterMethod)
   shimmer.wrap(Router.prototype, 'route', wrapRouterMethod)
 
   return Router
 })
 
+const queryParserReadCh = channel('datadog:query:read:finish')
+
+addHook({ name: 'router', versions: ['>=2'] }, Router => {
+  const WrappedRouter = shimmer.wrapFunction(Router, function (originalRouter) {
+    return function wrappedMethod (options) {
+      const router = originalRouter.call(this, options)
+
+      if (!router._queryParsingWrapped) {
```

Comment:
Do not modify router object, use weakSet instead

Reply:
@uurien after checking express 5 codebase, we do call handle function only once not at each use(...)
I removed `_queryParsingWrapped` and seems to be good